src/historywork/WriteVerifiedCheckpointHashesWork.cpp

```diff
@@ -14,10 +15,45 @@
 #include "work/ConditionalWork.h"
 #include <Tracy.hpp>
 #include <algorithm>
+#include <filesystem>
 #include <fmt/format.h>
 
 namespace stellar
 {
+LedgerNumHashPair
+WriteVerifiedCheckpointHashesWork::loadLatestHashPairFromJsonOutput(
+    std::string const& filename)
+{
+    if (!std::filesystem::exists(filename))
+    {
+        // If the file does not exist, the latest ledger is genesis.
+        return {LedgerManager::GENESIS_LEDGER_SEQ, Hash{}};
+    }
+
+    std::ifstream in(filename);
```

Comment:
Should we specify the openmode std::ios::out since we would be appending to the file.